code/modules/mod/mod_core.dm

```diff
@@ -79,9 +79,14 @@
 	/// Installed cell.
 	var/obj/item/stock_parts/cell/cell
 
+/obj/item/mod/core/standard/handle_atom_del(atom/deleting_atom)
+	if(deleting_atom == cell)
+		cell = null
+	return ..()
+
+
 /obj/item/mod/core/standard/Destroy()
-	if(cell)
-		QDEL_NULL(cell)
+	QDEL_NULL(cell)
```

Comment:
are you sure qdel(null) doesn't runtime?

Reply:
100%.